### PR TITLE
Fix fancy prompt prompt color

### PR DESCRIPTION
**Pull request by TibbTobb**

### CODE REVIEW

File: fancy_prompt/fancy_prompt.py [hunk 1]
```diff
@@ -35,8 +35,10 @@ def invoke(argument, from_tty):
         global colour_enabled
         if argument == 'on':
             colour_enabled = True
+            return
         elif argument == 'off':
             colour_enabled = False
+            return
```
Comment (barisione) [inline fancy_prompt/fancy_prompt.py]: This is a small style thing but I think it's clearer in cases like this if it's:
```
if ...:
    foo()
elif ...:
    bar()
else:
    raise ValueError()
```

This is because the `raise` is an alternative to the others.

I prefer early returns when it's some condition checking for instance:
```
if something_is_not_ok:
    return:

if cannot_do_something:
    raise Foo()

... # normal code.
```

File: fancy_prompt/fancy_prompt.py [hunk 2]
```diff
@@ -185,7 +187,7 @@ def prompt_hook(current_prompt):
     prompt_extra = ' '.join(fn() for fn in prompt_components)
 
     # Interpolate extra components into our base prompt string.
-    return '(udb {}) '.format(prompt_extra)
+    return '(udb {}) '.format(gdb.prompt.substitute_prompt(prompt_extra))
```
Comment (barisione) [inline fancy_prompt/fancy_prompt.py]: I didn't know this existed! Where did you find it?

Did you look at the code?
It looks like `\[` just gets translated into `\001` and `\]` into `\002`.

File: fancy_prompt/fancy_prompt.py [hunk 3]
```diff
@@ -109,14 +109,14 @@ def undodb_get_time():
...
-TERM_CYAN = r'\[\033[36m\]'
-TERM_RESET = r'\[\033[m\]'
+TERM_BRIGHT_WHITE = '\\[\033[1m\\]'
+TERM_BRIGHT_YELLOW = '\\[\033[1;33m\\]'
+TERM_BRIGHT_GREEN = '\\[\033[1;32m\\]'
+TERM_BRIGHT_RED = '\\[\033[1;31m\\]'
+TERM_BRIGHT_BLUE = '\\[\033[1;34m\\]'
+TERM_MAGENTA = '\\[\033[35m\\]'
+TERM_CYAN = '\\[\033[36m\\]'
+TERM_RESET = '\\[\033[m\\]'
```
Comment (barisione) [inline fancy_prompt/fancy_prompt.py]: Now that you use `gdb.prompt.substitute_prompt`, what happens if you revert this change?

File: fancy_prompt/fancy_prompt.py (hunk 2 above)
Comment (TibbTobb) [reply fancy_prompt/fancy_prompt.py]: It was in the original code. I removed it because I didn't think it was needed.

File: fancy_prompt/fancy_prompt.py (hunk 1 above)
Comment (TibbTobb) [reply fancy_prompt/fancy_prompt.py]: Fixed

File: fancy_prompt/fancy_prompt.py [hunk 4]
```diff
@@ -35,9 +35,11 @@ def invoke(argument, from_tty):
         global colour_enabled
         if argument == 'on':
             colour_enabled = True
+            return
```
Comment (barisione) [inline fancy_prompt/fancy_prompt.py]: This `return` is not needed any more.

**Comment by barisione** [comment]
Ping?

File: fancy_prompt/fancy_prompt.py (hunk 3 above)
Comment (TibbTobb) [reply fancy_prompt/fancy_prompt.py]: The original version with raw strings still doesn't work. Without raw strings it works but then pylint complains about 'Anomalous backslash in string'.

File: fancy_prompt/fancy_prompt.py (hunk 4 above)
Comment (TibbTobb) [reply fancy_prompt/fancy_prompt.py]: Fixed
